### PR TITLE
added changelogDestination & invocationMethod

### PR DESCRIPTION
**Pull request by danielsinai**

### CODE REVIEW

File: examples/resources/port-labs_action/main.tf [hunk 1]
```diff
@@ -4,7 +4,10 @@ resource "port-labs_action" "restart_microservice" {
 		identifier = "restart-micrservice"
 		blueprint_identifier = port-labs_blueprint.microservice.identifier
 		trigger = "DAY-2"
-		invocation_method = "KAFKA"
+		invocation_method = {
+			"type" = "WEBHOOK"
+			"url" = "https://app.getport.io"
```
Comment (hedwigz) [inline examples/resources/port-labs_action/main.tf:9]: wrong syntax

File: port/cli/models.go [hunk 2]
```diff
@@ -43,26 +43,37 @@ type (
 		Required   []string                     `json:"required,omitempty"`
 	}
 
+	ActionInvocationMethod struct {
+		Type string `json:"type,omitempty"`
+		Url  string `json:"url,omitempty"`
+	}
+
+	BlueprintChangelogDesintaion struct {
```
Comment (hedwigz) [inline port/cli/models.go:51]: We can name it `ChangelogDestination` since it already is a field of `Blueprint`

File: port/cli/models.go [hunk 3]
```diff
@@ -43,26 +43,37 @@ type (
 		Required   []string                     `json:"required,omitempty"`
 	}
 
+	ActionInvocationMethod struct {
```
Comment (hedwigz) [inline port/cli/models.go:46]: ditto

File: port/resource_port_action.go [hunk 4]
```diff
@@ -156,7 +170,14 @@ func actionResourceToBody(d *schema.ResourceData) (*cli.Action, error) {
 	action.Title = d.Get("title").(string)
 	action.Icon = d.Get("icon").(string)
 	action.Description = d.Get("description").(string)
-	action.InvocationMethod = d.Get("invocation_method").(string)
+	if invocation_method, ok := d.GetOk("invocation_method"); ok {
```
Comment (hedwigz) [inline port/resource_port_action.go:173]:
```suggestion
	if invocationMethod, ok := d.GetOk("invocation_method"); ok {
```
golang variables are commonly camelCased

File: port/resource_port_action.go [hunk 5]
```diff
@@ -156,7 +170,14 @@ func actionResourceToBody(d *schema.ResourceData) (*cli.Action, error) {
...
+		if action.InvocationMethod == nil {
+			action.InvocationMethod = &cli.ActionInvocationMethod{}
+		}
+
+		action.InvocationMethod.Type = invocation_method.(*schema.Set).List()[0].(map[string]interface{})["type"].(string)
```
Comment (hedwigz) [inline port/resource_port_action.go:178]: Maybe this can look tidier somehow 🤔

File: port/resource_port_blueprint.go [hunk 6]
```diff
@@ -191,8 +211,17 @@ func blueprintResourceToBody(d *schema.ResourceData) (*cli.Blueprint, error) {
 
 	b.Title = d.Get("title").(string)
 	b.Icon = d.Get("icon").(string)
-
 	props := d.Get("properties").(*schema.Set)
+
+	if changelog_destination, ok := d.GetOk("changelog_destination"); ok {
```
Comment (hedwigz) [inline port/resource_port_blueprint.go:216]: ditto

File: port/resource_port_blueprint.go [hunk 7]
```diff
@@ -191,8 +211,17 @@ func blueprintResourceToBody(d *schema.ResourceData) (*cli.Blueprint, error) {
...
+		if b.ChangelogDestination == nil {
+			b.ChangelogDestination = &cli.BlueprintChangelogDesintaion{}
+		}
+
+		b.ChangelogDestination.Type = changelog_destination.(*schema.Set).List()[0].(map[string]interface{})["type"].(string)
+		b.ChangelogDestination.Url = changelog_destination.(*schema.Set).List()[0].(map[string]interface{})["url"].(string)
```
Comment (hedwigz) [inline port/resource_port_blueprint.go:222]: ditto

**Review by hedwigz — APPROVED** [review]
🚀

**Review by hedwigz — CHANGES_REQUESTED** [review]
🚀  looking good!  
please run `make gen-docs` and then i'll approve

**Comment by danielsinai** [comment]
> 🚀 looking good! please run `make gen-docs` and then i'll approve

Thanks :) 
reGenerated the docs

**Review by hedwigz — APPROVED** [review]

**Pull request merged**